10up/ElasticPress
# refactor meta key allowed check

### Description of the Change

When a meta key is deleted from a post, this does NOT trigger reindex of that post due to the way of allowed metas are detected.

This change makes it so that we have the logic that evaluates if given meta_key is allowed separate from the logic that prepares metas for indexing. That way we can trigger even for meta key that is not being prepared to indexed (because it was just deleted) but is allowed to be in index (was in the indexed document before).


### Alternate Designs



### Possible Drawbacks



### Verification Process

(sorry the steps are specific to our dev-env, but I guess they still make sense?)

1) Add this filter impl:
```
add_filter( 'vip_search_post_meta_allow_list', function ( $allow, $post = null ) {
	$allow['testing'] = true;
	return $allow;
}, 10, 2 );
```
2) Add meta - `vip dev-env exec --  wp post meta set 1 testing foobar`
3) Check the document in ES - `vip dev-env exec --  wp vip-search documents get post 1 --format=json | jq '.[0].meta'`
4) Remove meta `vip dev-env exec --  wp  post meta delete 1 testing`
5) Check the document in ES - `vip dev-env exec --  wp vip-search documents get post 1 --format=json | jq '.[0].meta'`



### Checklist:






### Changelog Entry

Fixed - Post reindex on meta deletion

### Credits

@pschoffer
